File: examples/Perl/msgqueue.pl
# Simple message queuing broker in Perl
# Same as request-reply broker but using shared queue proxy

use strict;
use warnings;
use v5.10;

use ZMQ::FFI;
use ZMQ::FFI::Constants qw(ZMQ_ROUTER ZMQ_DEALER);

my $context = ZMQ::FFI->new();

# Socket facing clients
my $frontend = $context->socket(ZMQ_ROUTER);
$frontend->bind('tcp://*:5559');

# Socket facing services
my $backend = $context->socket(ZMQ_DEALER);
$backend->bind('tcp://*:5560');

# Start the proxy
$context->proxy($frontend, $backend);

# We never get here...

File: examples/C++/msgqueue.cpp
//
//  Simple message queuing broker in C++
//  Same as request-reply broker but using QUEUE device
//
// Olivier Chamoux <olivier.chamoux@fr.thalesgroup.com>


#include "zhelpers.hpp"

int main (int argc, char *argv[])
{
    zmq::context_t context(1);

    //  Socket facing clients
    zmq::socket_t frontend (context, ZMQ_ROUTER);
    frontend.bind("tcp://*:5559");

    //  Socket facing services
    zmq::socket_t backend (context, ZMQ_DEALER);
    backend.bind("tcp://*:5560");

    //  Start the proxy
    zmq::proxy(frontend, backend, nullptr);
    return 0;
}
